src/SFA.DAS.Campaign.Web/Views/Shared/Components/TrainingProviderHeading/Details.cshtml

```diff
@@ -1,12 +1,36 @@
-﻿@model Sfa.Das.Sas.Shared.Components.ViewModels.TrainingProviderDetailQueryViewModel
+﻿@using Microsoft.AspNetCore.Html
+@model Sfa.Das.Sas.Shared.Components.ViewModels.TrainingProviderDetailQueryViewModel
+
+@{
+    var titleModifierClass = "";
+
+    System.Text.Encodings.Web.HtmlEncoder hEncoder = System.Text.Encodings.Web.HtmlEncoder.Default;
+
+    var pageHeading = new System.IO.StringWriter();
+
+    Component.InvokeAsync("TrainingProviderTitle", new { searchQueryModel = Model }).Result.WriteTo(pageHeading, hEncoder);
+
+    var pageHeadingStr = new HtmlContentBuilder().AppendHtml(pageHeading.ToString().Trim());
+
+    var pageHeadingLength = pageHeading.ToString().Trim().Length;
+
+    if (pageHeadingLength > 20)
```

Comment:
I think if we have more occurrences of this logic we should probably abstract out in some way, partial view or something.

Reply:
Agreed